src/test/java/com/wikia/webdriver/elements/mercury/components/TopBar.java

```diff
@@ -30,23 +41,87 @@ public TopBar(WebDriver driver) {
 
   public Navigation openNavigation() {
     PageObjectLogging.logInfo("Open navigation");
-    wait.forElementClickable(openNavButton);
-    openNavButton.click();
+    wait.forElementClickable(hamburgerIcon);
+    hamburgerIcon.click();
 
     PageObjectLogging.logInfo("Navigation is opened");
-    wait.forElementVisible(navigationComponent);
+    wait.forElementVisible(navMenu);
 
     return new Navigation(driver);
   }
 
-  public Navigation closeNavigation() {
-    PageObjectLogging.logInfo("Close navigation");
-    wait.forElementClickable(closeNavButton);
-    closeNavButton.click();
+  public Search openSearch() {
+    PageObjectLogging.logInfo("Open search");
+    wait.forElementClickable(searchIcon);
+    searchIcon.click();
 
-    PageObjectLogging.logInfo("Navigation is closed");
-    wait.forElementNotVisible(navigationComponent);
+    PageObjectLogging.logInfo("Search is opened");
+    wait.forElementVisible(navigationComponent);
+
+    return new Search(driver);
+  }
+
+  public Navigation clickCloseButton() {
+    PageObjectLogging.logInfo("Click close button");
+    wait.forElementClickable(closeButton);
+    closeButton.click();
 
     return new Navigation(driver);
   }
+
+  public boolean isNavigationBarVisible() {
+    try {
+      wait.forElementVisible(navBar);
+      PageObjectLogging.logInfo("Navigation bar is visible");
+      return true;
+    } catch (Exception e) {
+      PageObjectLogging.logInfo("Navigation bar is not visible");
+      return false;
+    }
+  }
+
+  public boolean isLogoVisible() {
+    try {
+      wait.forElementVisible(logo);
+      PageObjectLogging.logInfo("Wikia logo is visible");
+      return true;
+    } catch (Exception e) {
+      PageObjectLogging.logInfo("Wikia logo is not visible");
+      return false;
+    }
+  }
+
+  public boolean isHamburgerIconVisible() {
+    try {
+      wait.forElementVisible(hamburgerIcon);
+      PageObjectLogging.logInfo("Hamburger icon is visible");
+      return true;
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)